src/Platform/Microsoft.Testing.Platform/Logging/InMemoryStoreLogger.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Concurrent;
+
+namespace Microsoft.Testing.Platform.Logging;
+
+internal abstract class InMemoryStoreLogger<T> : ILogger
+    where T : class
+{
+    private readonly LogLevel _logLevel;
+
+    public InMemoryStoreLogger(LogLevel logLevel)
+    {
+        _logLevel = logLevel;
+    }
+
+    public ConcurrentBag<T> Values { get; private set; } = new();
```

Comment:
Shall it be only protected?

Reply:
mmm needs to be accessed from outside it's sealed, what you mean?

Reply:
like `IEnum<T>` on the type and make it protected?

Reply:
We can also remove this type and implement it only for the server mode...maybe it's too much have this abstraction.